### PR TITLE
Adding new apis

### PR DESCRIPTION
**Pull request by MiriBerezin**
BulkCollectionsAction
QueryApprovedPlans

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [V] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [V] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [V] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [V] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [V] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [V] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [x] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by jianyexi** [comment]
> Hi, @MiriBerezin Miri Berezin FTE Thanks for your PR. I am workflow bot for review process. Here are some small tips.
> * Please ensure to do self-check against checklists in first PR comment.
> * PR assignee is the person auto-assigned and responsible for your current PR reviewing and merging.
> * For specs comparison cross API versions,  Use [API Specs Comparison Report Generator](https://portal.azure-devex-tools.com/tools/diff?pr=Azure/azure-rest-api-specs/14910)
> * If there is CI failure(s), to fix CI error(s) is mandatory for PR merging; or you need to provide justification in PR comment for explanation. [How to fix?](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/ci-fix.md)
> 
> Any feedback about review process or workflow bot, pls contact swagger and tools team. [vsswagger@microsoft.com](mailto:vsswagger@microsoft.com)

@MiriBerezin  please follow the tips in the comment

**Comment by MiriBerezin** [comment]
@jianyexi I posted a request for breaking changes. 

Can explain here as well: We are still in development mode and we need these two APIs. 

[https://msazure.visualstudio.com/One/_workitems/edit/10195585](url)

Please let me know if there are further steps I need to take

**Comment by gupele** [comment]
Do not merge, thanks

**Comment by MiriBerezin** [comment]
We are ready to merge after review. Thanks!

**Comment by pilor** [comment]
Please go through and complete the checklist in the PR description

**Comment by MiriBerezin** [comment]
@pilor, done

**Comment by pilor** [comment]
2021-06-01 is not in the RP manifest so updates to this version are OK

File: specification/marketplace/resource-manager/Microsoft.Marketplace/stable/2021-06-01/Marketplace.json
Comment (pilor) [inline specification/marketplace/resource-manager/Microsoft.Marketplace/stable/2021-06-01/Marketplace.json:2208]: This will be easier to consume if its an array of plan objects with planId as a property in each object. Why does it need to be a dictionary with dynamic keys?

File: specification/marketplace/resource-manager/Microsoft.Marketplace/stable/2021-06-01/Marketplace.json
Comment (pilor) [inline specification/marketplace/resource-manager/Microsoft.Marketplace/stable/2021-06-01/Marketplace.json:530]: Why does this need a bulk operation instead of performing the operation on each collection individually?  Bulk operations are problematic because they may perform actions the user did not have RBAC rights to perform (since the operation is not performed directly on the scope it affects) and they don't expose in the activity log everything they are modifying.  It also affects downstream notifications of changes, etc...

File: specification/marketplace/resource-manager/Microsoft.Marketplace/stable/2021-06-01/Marketplace.json
Comment (MiriBerezin) [reply specification/marketplace/resource-manager/Microsoft.Marketplace/stable/2021-06-01/Marketplace.json:2208]: We want to enforce the plan id uniqueness in the result

File: specification/marketplace/resource-manager/Microsoft.Marketplace/stable/2021-06-01/Marketplace.json
Comment (MiriBerezin) [reply specification/marketplace/resource-manager/Microsoft.Marketplace/stable/2021-06-01/Marketplace.json:530]: Currently, we don't have RBAC per collection. It is on post Ni roadmap.
Once we get there, we have 2 options ( on the RP side): 
1. Check RBAC  for each collection and fail the action if there is a least one collection that is not authorized.
2. Check RBAC  for each collection and activate only for the allowed ones. In this case, the response will contain the details for the ones that passed and the ones that failed. The current response already supports this scenario.

File: specification/marketplace/resource-manager/Microsoft.Marketplace/stable/2021-06-01/Marketplace.json
Comment (pilor) [reply specification/marketplace/resource-manager/Microsoft.Marketplace/stable/2021-06-01/Marketplace.json:2208]: You control the response so you can enforce uniqueness regardless of the response schema.  If that is the only reason please use a strict schema instead of dynamic keys

File: specification/marketplace/resource-manager/Microsoft.Marketplace/stable/2021-06-01/Marketplace.json
Comment (MiriBerezin) [reply specification/marketplace/resource-manager/Microsoft.Marketplace/stable/2021-06-01/Marketplace.json:530]: We have a business requirement to perform an action delete/enable/disable for all collections in a click.
On the server side, we have a transaction(DB stored procedure) performing that operation. If we don't do that in bulk we get conflicts and high-performance impact and not acceptable latency.
 Thus the API must be a bulk operation.

File: specification/marketplace/resource-manager/Microsoft.Marketplace/stable/2021-06-01/Marketplace.json
Comment (MiriBerezin) [reply specification/marketplace/resource-manager/Microsoft.Marketplace/stable/2021-06-01/Marketplace.json:2208]: Fixed to list of objects

**Comment by MiriBerezin** [comment]
@jianyexi we got arm sign-off we would like to merge to master

**Comment by ruowan** [comment]
No deploy api-version 2021-06-01. Approve this breakingChange.

**Review by ruowan — APPROVED** [review]

**Pull request merged**
